uReplicator-Common/src/main/java/com/uber/stream/kafka/mirrormaker/common/utils/NetUtils.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2015-2019 Uber Technologies, Inc. (streaming-data@uber.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.uber.stream.kafka.mirrormaker.common.utils;
+
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Enumeration;
+
+
+/**
+ * @author Tboy
+ */
+public class NetUtils {
+
+  public static String getFirstNoLoopbackIP4Address() {
```

Comment:
what's the problem this pr aims to resolve? please provide more details 
also, please provide test plan

Reply:
if we using uReplicator in federated mode, controller has to register itself using hostname. this is ok when we in physical env.  but if in docker, hostname maybe like "ureplicator-manager-5b7d66988d-9j7hk ",  rest service is not working anymore。